File: 2006-count-number-of-pairs-with-absolute-difference-k/2006-count-number-of-pairs-with-absolute-difference-k.cpp
class Solution {
public:
    int countKDifference(vector<int>& nums, int k) {
        unordered_map<int,int>mp;
        int cnt=0;
        for(auto it:nums){
            cnt+=(mp[it-k]+mp[it+k]);
            mp[it]++;
        }
        return cnt;
    }
};

File: 2006-count-number-of-pairs-with-absolute-difference-k/2006-count-number-of-pairs-with-absolute-difference-k.java
class Solution {
    public int countKDifference(int[] arr, int k) {
        int n=arr.length;
        int count = 0;
        for(int i = 0; i < n; i++)
        {
            int x = arr[i];
            for(int j = 0; j < n; j++)
            {
                if(arr[j] == (x+k) || arr[j] == (x-k))
                {
                    count++;
                }
            }
        }
        return count/2;
    }
}